pkg/gateway/operations/base.go

```diff
@@ -40,6 +40,10 @@ const (
 
 type ActionIncr func(action, repository, ref, userID string)
 
+type UploadPathProvider interface {
```

Comment:
Why is the UploadPathProvider interface defined under the gateway package?

Reply:
Local interface that match the requirements from path provider.